src/sentry/utils/linksign.py

```diff
@@ -0,0 +1,61 @@
+from django.core import signing
+from django.core.urlresolvers import reverse
+
+from sentry import options
+from sentry.models import User
+from sentry.utils.http import absolute_uri
+from sentry.utils.numbers import base36_encode, base32_decode
+
+
+def get_signer():
+    return signing.TimestampSigner(salt='sentry-link-signature')
+
+
+def generate_signed_link(user, viewname, args=None, kwargs=None):
+    """This returns an absolute URL where the given user is signed in for
+    the given viewname with args and kwargs.  This returns a redirect link
+    that if followed sends the user to another URL which carries another
+    signature that is valid for that URL only.  The user can also be a user
+    ID.
+    """
+    if hasattr(user, 'is_authenticated'):
+        if not user.is_authenticated():
+            raise RuntimeError('Need an authenticated user to sign a link.')
+        user_id = user.id
+    else:
+        user_id = user
+
+    path = reverse(viewname, args=args, kwargs=kwargs)
+    item = '%s|%s|%s' % (options.get('system.url-prefix'), path, user_id)
+    signature = ':'.join(get_signer().sign(item).rsplit(':', 2)[1:])
```

Comment:
It looks like `signature` would work here to avoid splitting: https://github.com/django/django/blob/stable/1.6.x/django/core/signing.py#L156-L159